### PR TITLE
fix: Enable more linters

### PR DESCRIPTION
**Pull request by mateusoliveira43**
Enable more strict checks with Go linters using golangci-lint.

### CODE REVIEW

File: .golangci.yml [hunk 1]
```diff
@@ -1,40 +1,128 @@
+# Documentation reference https://github.com/golangci/golangci-lint/blob/v1.56.2/.golangci.reference.yml
```
Comment (mateusoliveira43) [inline .golangci.yml:1]: update version used

File: .golangci.yml [hunk 2]
```diff
@@ -1,40 +1,128 @@
...
 run:
-  deadline: 5m
+  skip-dirs-use-default: false
+  modules-download-mode: readonly
   allow-parallel-runners: true
+  skip-dirs:
```
Comment (mateusoliveira43) [inline .golangci.yml:6]: will remove these skips prior to merge

File: .golangci.yml [hunk 3]
```diff
@@ -1,40 +1,128 @@
...
+    - linters:
+        - revive
+      text: "^struct-tag: unknown option 'inline' in JSON tag$"
+    - linters:
+        - revive
+      text: "^add-constant: avoid magic numbers like '0', create a named constant for it$"
+    - linters:
+        - revive
+      text: "^add-constant: avoid magic numbers like '1', create a named constant for it$"
+    # - linters:
```
Comment (mateusoliveira43) [inline .golangci.yml:120]: will uncomment or remove prior to merge

File: .golangci.yml (hunk 2 above)
Comment (mateusoliveira43) [reply .golangci.yml:6]: will let this one for another PR

we are not using e2e yet

File: internal/common/function/function.go [hunk 4]
```diff
@@ -0,0 +1,224 @@
...
+	if nonAdminBackup.ObjectMeta.UID != types.UID(nabOriginUUID) {
+		return nil, fmt.Errorf("UID from annotation does not match UID of fetched NonAdminBackup object")
+	}
+
+	return nonAdminBackup, nil
+}
+
+// TODO import?
+// Similar to as pkg/common/common.go:AppendUniqueKeyTOfTMaps from github.com/openshift/oadp-operator
+func mergeUniqueKeyTOfTMaps[T comparable](userMap ...map[T]T) (map[T]T, error) {
```
Comment (kaovilai) [inline internal/common/function/function.go:203]: This function needs more comment to quickly understand if latter maps overrides the previous one or vice versa.

File: internal/common/function/function.go
Comment (mateusoliveira43) [inline internal/common/function/function.go:1]: here we could separate in more files, but did not focused on this in this PR

File: internal/common/function/function.go (hunk 4 above)
Comment (mateusoliveira43) [reply internal/common/function/function.go]: done

**Comment by mpryc** [comment]
Thanks.

**Pull request merged**